clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -155,6 +156,33 @@ public void start() throws Exception {
         logger.log(LogLevel.Information, "HubConnected started.");
     }
 
+    /**
+     * Starts a connection to the server asynchronously.
+     * @throws Exception An error occurred while connecting.
+     */
+    public CompletableFuture startAsync() throws Exception {
```

Comment:
Is the convention to use the async suffix in java?

Reply:
I'm not entirely sure. For example, the [Azure Key Vault Client](https://github.com/Azure/azure-keyvault-java/blob/dev/azure-keyvault/src/main/java/com/microsoft/azure/keyvault/KeyVaultClientCustom.java#L152) and [Azure Service Bus Client](https://github.com/Azure/azure-service-bus-java/blob/dev/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/QueueClient.java#L409) both do this, and has sync/async versions using the `Async` suffix to distinguish them. I presume other Azure SDKs do it, but I haven't checked them all.

However, the `Async` suffix [has a different meaning in `CompletableFuture` itself](https://stackoverflow.com/a/47489654/29813). In that case, `thenApply` indicates that the continuation should be run in-line on the same thread, whereas `thenApplyAsync` indicates that the continuation should be run on a new thread-pool thread.

Reply:
It doesn't look like it, but I'm just prototyping here so I didn't want to mess with the existing APIs